pkg/trace/call.go

```diff
@@ -93,12 +186,27 @@ func EndCall(ctx context.Context) {
 	}
 
 	addDefaultTracerInfo(ctx, info)
-	info.ReportLatency()
-	traceInfo := log.F{"honeycomb.trace_id": ID(ctx), "event_name": "trace"}
+
+	if info.kind == metrics.CallKindExternal {
+		info.ReportOutboundLatency()
+	} else {
+		switch info.name { //nolint:exhaustive //Why: we only report latency metrics in this case on HTTP/gRPC call types.
+		case CallTypeHTTP:
+			info.ReportHTTPLatency()
+		case CallTypeGRPC:
+			info.ReportGRPCLatency()
+		}
```

Comment:
You could probably add a `callType.reportLatency(info)` and move the switch over to there.

Reply:
nice idea, i'll do that